In [1]:
import pandas as pd
import numpy as np

In [2]:
twitch_streams = pd.read_csv("Most_watched_games_on_Twitch.csv")
video_game_sales = pd.read_csv("vgsales.csv")
steam_data = pd.read_csv("steam_games.csv")

In [3]:
twitch_streams.head()

Unnamed: 0  Unnamed: 1  \
0                 League of Legends           1   
1                     Just Chatting           2   
2                          Fortnite           3   
3                Grand Theft Auto V           4   
4  Counter-Strike: Global Offensive           5   

                               Game   Watch time  Stream time  Peak viewers  \
0                 League of Legends  77530931640   1862302140       1782765   
1                     Just Chatting  72730827240   1107578340        839576   
2                          Fortnite  57041617200   4002480600       2331987   
3                Grand Theft Auto V  41795815320    754766340        389147   
4  Counter-Strike: Global Offensive  38520066780    880832640        828949   

   Peak channels  Streamers  Average viewers  Average channels  \
0           8802    1053035           147509              3543   
1          12126    1168725           138376              2107   
2         115814    2933441           108526              7615   
3           5847     773539            79520              1436   
4           5542     918769            73287              1675   

   Average viewer ratio  Followers gained  Views gained       FPR        VPR  \
0                 41.63          35542269    1568659867  1.145107  50.539378   
1                 65.67          81340586    1718298124  4.406402  93.084059   
2                 14.25         105529900    1047550469  1.581967  15.703518   
3                 55.38          28405862     668023673  2.258118  53.104409   
4                 43.73          23290831     916342710  1.586510  62.418852   

                        Unnamed: 15  
0                 League of Legends  
1                     Just Chatting  
2                          Fortnite  
3                Grand Theft Auto V  
4  Counter-Strike: Global Offensive

In [4]:
video_game_sales.head(50)

Rank                                          Name Platform    Year  \
0      1                                    Wii Sports      Wii  2006.0   
1      2                             Super Mario Bros.      NES  1985.0   
2      3                                Mario Kart Wii      Wii  2008.0   
3      4                             Wii Sports Resort      Wii  2009.0   
4      5                      Pokemon Red/Pokemon Blue       GB  1996.0   
5      6                                        Tetris       GB  1989.0   
6      7                         New Super Mario Bros.       DS  2006.0   
7      8                                      Wii Play      Wii  2006.0   
8      9                     New Super Mario Bros. Wii      Wii  2009.0   
9     10                                     Duck Hunt      NES  1984.0   
10    11                                    Nintendogs       DS  2005.0   
11    12                                 Mario Kart DS       DS  2005.0   
12    13                   Pokemon Gold/Pokemon Silver       GB  1999.0   
13    14                                       Wii Fit      Wii  2007.0   
14    15                                  Wii Fit Plus      Wii  2009.0   
15    16                            Kinect Adventures!     X360  2010.0   
16    17                            Grand Theft Auto V      PS3  2013.0   
17    18                 Grand Theft Auto: San Andreas      PS2  2004.0   
18    19                             Super Mario World     SNES  1990.0   
19    20  Brain Age: Train Your Brain in Minutes a Day       DS  2005.0   
20    21                 Pokemon Diamond/Pokemon Pearl       DS  2006.0   
21    22                              Super Mario Land       GB  1989.0   
22    23                           Super Mario Bros. 3      NES  1988.0   
23    24                            Grand Theft Auto V     X360  2013.0   
24    25                   Grand Theft Auto: Vice City      PS2  2002.0   
25    26                 Pokemon Ruby/Pokemon Sapphire      GBA  2002.0   
26    27                   Pokemon Black/Pokemon White       DS  2010.0   
27    28   Brain Age 2: More Training in Minutes a Day       DS  2005.0   
28    29                        Gran Turismo 3: A-Spec      PS2  2001.0   
29    30                Call of Duty: Modern Warfare 3     X360  2011.0   
30    31       Pokémon Yellow: Special Pikachu Edition       GB  1998.0   
31    32                       Call of Duty: Black Ops     X360  2010.0   
32    33                           Pokemon X/Pokemon Y      3DS  2013.0   
33    34                     Call of Duty: Black Ops 3      PS4  2015.0   
34    35                    Call of Duty: Black Ops II      PS3  2012.0   
35    36                    Call of Duty: Black Ops II     X360  2012.0   
36    37                Call of Duty: Modern Warfare 2     X360  2009.0   
37    38                Call of Duty: Modern Warfare 3      PS3  2011.0   
38    39                          Grand Theft Auto III      PS2  2001.0   
39    40                       Super Smash Bros. Brawl      Wii  2008.0   
40    41                       Call of Duty: Black Ops      PS3  2010.0   
41    42                   Animal Crossing: Wild World       DS  2005.0   
42    43                                  Mario Kart 7      3DS  2011.0   
43    44                                        Halo 3     X360  2007.0   
44    45                            Grand Theft Auto V      PS4  2014.0   
45    46          Pokemon HeartGold/Pokemon SoulSilver       DS  2009.0   
46    47                                Super Mario 64      N64  1996.0   
47    48                                Gran Turismo 4      PS2  2004.0   
48    49                            Super Mario Galaxy      Wii  2007.0   
49    50     Pokemon Omega Ruby/Pokemon Alpha Sapphire      3DS  2014.0   

           Genre                    Publisher  NA_Sales  EU_Sales  JP_Sales  \
0         Sports                     Nintendo     41.49     29.02      3.77   
1       Platform    

In [5]:
steam_data.head(50)

url   types  \
0     https://store.steampowered.com/app/379720/DOOM/     app   
1   https://store.steampowered.com/app/578080/PLAY...     app   
2   https://store.steampowered.com/app/637090/BATT...     app   
3     https://store.steampowered.com/app/221100/DayZ/     app   
4   https://store.steampowered.com/app/8500/EVE_On...     app   
5   https://store.steampowered.com/bundle/5699/Gra...  bundle   
6   https://store.steampowered.com/app/601150/Devi...     app   
7   https://store.steampowered.com/app/477160/Huma...     app   
8   https://store.steampowered.com/app/644930/They...     app   
9   https://store.steampowered.com/app/774241/Warh...     app   
10  https://store.steampowered.com/app/527230/For_...     app   
11  https://store.steampowered.com/app/567640/Dang...     app   
12    https://store.steampowered.com/app/323370/TERA/     app   
13  https://store.steampowered.com/app/393080/Call...     app   
14  https://store.steampowered.com/app/253250/Ston...     app   
15  https://store.steampowered.com/bundle/5641/Hea...  bundle   
16  https://store.steampowered.com/app/597170/Clon...     app   
17  https://store.steampowered.com/app/899440/GOD_...     app   
18  https://store.steampowered.com/app/767560/War_...     app   
19  https://store.steampowered.com/app/459220/Halo...     app   
20  https://store.steampowered.com/app/42700/Call_...     app   
21  https://store.steampowered.com/app/787480/Phoe...     app   
22  https://store.steampowered.com/app/785260/Team...     app   
23  https://store.steampowered.com/app/475150/Tita...     app   
24  https://store.steampowered.com/app/12210/Grand...     app   
25  https://store.steampowered.com/bundle/234/Port...  bundle   
26     https://store.steampowered.com/app/400/Portal/     app   
27  https://store.steampowered.com/app/246900/Visc...     app   
28  https://store.steampowered.com/app/704450/Neve...     app   
29  https://store.steampowered.com/app/589360/Ni_n...     app   
30  https://store.steampowered.com/app/372000/Tree...     app   
31  https://store.steampowered.com/app/414340/Hell...     app   
32  https://store.steampowered.com/app/413420/Dang...     app   
33  https://store.steampowered.com/app/638970/Yaku...     app   
34  https://store.steampowered.com/app/641990/The_...     app   
35  https://store.steampowered.com/app/6060/Star_W...     app   
36  https://store.steampowered.com/app/799960/Wiza...     app   
37  https://store.steampowered.com/app/532210/Life...     app   
38  https://store.steampowered.com/app/102500/King...     app   
39          https://store.steampowered.com/sub/18397/     sub   
40  https://store.steampowered.com/app/56437/Warha...     app   
41  https://store.steampowered.com/app/1038810/GOD...     app   
42  https://store.steampowered.com/app/240/Counter...     app   
43  https://store.steampowered.com/app/268050/The_...     app   
44  https://store.steampowered.com/bundle/2405/The...  bundle   
45  https://store.steampowered.com/app/955050/Brig...     app   
46  https://store.steampowered.com/app/32470/STAR_...     app   
47  https://store.steampowered.com/app/614570/Dish...     app   
48  https://store.steampowered.com/app/310950/Stre...     app   
49  https://store.steampowered.com/app/397460/The_...     app   

                                                 name  \
0                                                DOOM   
1                       PLAYERUNKNOWN'S BATTLEGROUNDS   
2                                          BATTLETECH   
3                                                DayZ   
4                                          EVE Online   
5          Grand Theft Auto V: Premium Online Edition   
6                                     Devil May Cry 5   
7                                    Human: Fall Flat   
8                                   They Are Billions   
9                                Warhammer: Chaosbane   
10                                       For The King   
11                    Danganronpa V3: Killing Harm

In [6]:
steam_data.tail()

url types  \
40828  https://store.steampowered.com/app/899836/Rock...   app   
40829  https://store.steampowered.com/app/899832/Rock...   app   
40830  https://store.steampowered.com/app/906840/Fant...   app   
40831  https://store.steampowered.com/app/906635/Mega...   app   
40832  https://store.steampowered.com/app/908000/Stor...   app   

                                                    name  \
40828  Rocksmith® 2014 Edition – Remastered – Sabaton...   
40829  Rocksmith® 2014 Edition – Remastered – Stone T...   
40830  Fantasy Grounds - Quests of Doom 4: A Midnight...   
40831                       Mega Man X5 Sound Collection   
40832                                   Stories In Stone   

                                            desc_snippet recent_reviews  \
40828                                                NaN            NaN   
40829                                                NaN            NaN   
40830                                                NaN            NaN   
40831                                                NaN            NaN   
40832  An RPG about a tribe exploring an ancient worl...            NaN   

      all_reviews  release_date                developer  \
40828         NaN  Feb 12, 2019  Ubisoft - San Francisco   
40829         NaN   Feb 5, 2019  Ubisoft - San Francisco   
40830         NaN  Jul 31, 2018      SmiteWorks USA, LLC   
40831         NaN  Jul 24, 2018          CAPCOM CO., LTD   
40832         NaN   Aug 8, 2018      16 Bit Psych,Kyle B   

                             publisher  \
40828                              NaN   
40829                              NaN   
40830                              NaN   
40831  CAPCOM CO., LTD,CAPCOM CO., LTD   
40832        Self-Publish,Self-Publish   

                                            popular_tags  \
40828                                  Casual,Simulation   
40829                                  Casual,Simulation   
40830  RPG,Indie,Strategy,Software,Turn-Based,Fantasy...   
40831                                             Action   
40832                                      RPG,Adventure   

                                            game_details  \
40828  Single-player,Shared/Split Screen,Downloadable...   
40829  Single-player,Shared/Split Screen,Downloadable...   
40830  Multi-player,Co-op,Cross-Platform Multiplayer,...   
40831  Single-player,Downloadable Content,Steam Achie...   
40832  Single-player,Partial Controller Support,Profi...   

                                               languages  achievements  \
40828  English,German,French,Italian,Spanish - Spain,...           NaN   
40829  English,German,French,Italian,Spanish - Spain,...           NaN   
40830                                            English           NaN   
40831  English,French,Italian,German,Spanish - Spain,...           NaN   
40832                                            English           NaN   

                    genre                                   game_description  \
40828   Casual,Simulation   About This Content Play "Ghost Division" by S...   
40829   Casual,Simulation   About This Content Play "Trippin’ on a Hole i...   
40830  Indie,RPG,Strategy   About This Content  Quests of Doom 4: A Midni...   
40831              Action   About This Content Get equipped with the stun...   
40832       Adventure,RPG   About This Game  An RPG about a tribe explori...   

      mature_content                               minimum_requirements  \
40828            NaN  Minimum:,OS:,Windows Vista, Windows 7, Windows...   
40829            NaN  Minimum:,OS:,Windows Vista, Windows 7, Windows...   
40830            NaN  Minimum:,OS:,Windows 7x , 8x or 10x,Processor:...   
40831            NaN  Minimum:,OS:,WINDOWS® 7 (64bit),Processor:,Int...   
40832            NaN                                                NaN   

                                recommended_requirements original_price  \
40828  Recommended:,OS:,Windows Vista, Windows 7, Win...

In [7]:
len(steam_data)

40833

### With 40,833 rows, the "steam_data" dataset is large enough for us to obtain some interesting findings with our Machine Learning Model. We will need to drop a few columns though.

In [8]:
len(twitch_streams)

100

### The "twitch_streams" dataset only has 100 rows, which makes it VERY small dataset.

In [9]:
len(steam_data.columns)

20

In [10]:
len(video_game_sales)

16598

###  The "video_game_sales" dataset has 16,598 rows, which also makes it a good candidate for our modeling.
### Now, we must clean our "steam_data" by dropping any columns that contain information that we are not trying to analyze.

In [11]:
steam_data_new = steam_data.drop(["url", "types", "desc_snippet", "languages", "recent_reviews", "release_date", "developer", "publisher", "achievements", "game_description", "mature_content", "minimum_requirements", "recommended_requirements", "original_price", "discount_price"], axis=1)
steam_data_new.head()

name  \
0                           DOOM   
1  PLAYERUNKNOWN'S BATTLEGROUNDS   
2                     BATTLETECH   
3                           DayZ   
4                     EVE Online   

                                         all_reviews  \
0  Very Positive,(42,550),- 92% of the 42,550 use...   
1  Mixed,(836,608),- 49% of the 836,608 user revi...   
2  Mostly Positive,(7,030),- 71% of the 7,030 use...   
3  Mixed,(167,115),- 61% of the 167,115 user revi...   
4  Mostly Positive,(11,481),- 74% of the 11,481 u...   

                                        popular_tags  \
0  FPS,Gore,Action,Demons,Shooter,First-Person,Gr...   
1  Survival,Shooter,Multiplayer,Battle Royale,PvP...   
2  Mechs,Strategy,Turn-Based,Turn-Based Tactics,S...   
3  Survival,Zombies,Open World,Multiplayer,PvP,Ma...   
4  Space,Massively Multiplayer,Sci-fi,Sandbox,MMO...   

                                        game_details  \
0  Single-player,Multi-player,Co-op,Steam Achieve...   
1             Multi-player,Online Multi-Player,Stats   
2  Single-player,Multi-player,Online Multi-Player...   
3  Multi-player,Online Multi-Player,Steam Worksho...   
4  Multi-player,Online Multi-Player,MMO,Co-op,Onl...   

                                               genre  
0                                             Action  
1             Action,Adventure,Massively Multiplayer  
2                          Action,Adventure,Strategy  
3             Action,Adventure,Massively Multiplayer  
4  Action,Free to Play,Massively Multiplayer,RPG,...

In [12]:
steam_data_new.tail()

name all_reviews  \
40828  Rocksmith® 2014 Edition – Remastered – Sabaton...         NaN   
40829  Rocksmith® 2014 Edition – Remastered – Stone T...         NaN   
40830  Fantasy Grounds - Quests of Doom 4: A Midnight...         NaN   
40831                       Mega Man X5 Sound Collection         NaN   
40832                                   Stories In Stone         NaN   

                                            popular_tags  \
40828                                  Casual,Simulation   
40829                                  Casual,Simulation   
40830  RPG,Indie,Strategy,Software,Turn-Based,Fantasy...   
40831                                             Action   
40832                                      RPG,Adventure   

                                            game_details               genre  
40828  Single-player,Shared/Split Screen,Downloadable...   Casual,Simulation  
40829  Single-player,Shared/Split Screen,Downloadable...   Casual,Simulation  
40830  Multi-player,Co-op,Cross-Platform Multiplayer,...  Indie,RPG,Strategy  
40831  Single-player,Downloadable Content,Steam Achie...              Action  
40832  Single-player,Partial Controller Support,Profi...       Adventure,RPG

In [13]:
print(steam_data_new.columns)

Index(['name', 'all_reviews', 'popular_tags', 'game_details', 'genre'], dtype='object')


In [14]:
len(steam_data_new.columns)

5

### Download as csv file.

In [16]:
steam_data_new.to_csv('new_steam_data.csv')